### PR TITLE
feat: Add delete functionality

### LINKED ISSUE #84: [New API] Add Delete API
Our backend server now supports delete, this functionality needs to be added into the SDK.

Need to update to use newest proto: https://github.com/momentohq/client_protos/blob/main/proto/cacheclient.proto#L35

### PR DESCRIPTION
**Pull request by malandis**
Closes #84 

Expose delete method to the client. Update momento wire types, hook up gRPC to data client backend, and updates clients.

### CODE REVIEW

File: src/momento/aio/_scs_data_client.py [hunk 1]
```diff
@@ -234,5 +233,24 @@ async def multi_get(
             _momento_logger.debug(f"multi_get failed with response: {e}")
             raise _cache_service_errors_converter.convert(e)
 
+    async def delete(
+        self, cache_name: str, key: Union[str, bytes]
+    ) -> cache_sdk_ops.CacheDeleteResponse:
+        _validate_cache_name(cache_name)
+        try:
+            _momento_logger.debug(f"Issuing a delete request with key {str(key)}")
```
Comment (eaddingtonwhite) [inline src/momento/aio/_scs_data_client.py:241]: just as im looking at this noticed we could probably DRY up our access logging. Maybe for follow up pr but we could probably extract this debug logging out into some standard decorator vs writing out in every method.

File: src/momento/simple_cache_client.py [hunk 2]
```diff
@@ -269,6 +270,28 @@ def multi_get(
...
+
+        Returns:
+            CacheDeleteResponse
+
+        Raises:
+            InvalidArgumentError: If validation fails for provided method arguments.
+            BadRequestError: If the provided inputs are rejected by server because they are invalid
+            NotFoundError: If the cache with the given name doesn't exist.
+            AuthenticationError: If the provided Momento Auth Token is invalid.
+            InternalServerError: If server encountered an unknown error while trying to retrieve the item.
```
Comment (eaddingtonwhite) [inline src/momento/simple_cache_client.py:290]:
```suggestion
            InternalServerError: If server encountered an unknown error while trying to delete the item.
```

**Review by eaddingtonwhite — APPROVED** [review]
Few nits but looks straight forward @malandis nice work! 🚢

File: src/momento/aio/_scs_data_client.py (hunk 1 above)
Comment (kvcache) [reply src/momento/aio/_scs_data_client.py:241]: this is true.


Also, this strategy for logging is not great. You're (1) encoding, (2) copying the key and (3) allocating yet a 3rd copy of the memory just to throw it away probably 99.99999% of the time or more. Why does this need to do all of that work when the `debug` level is not enabled?

Comment (kvcache) [reply src/momento/aio/_scs_data_client.py:241]: oh geeze, here & everywhere i see.

Comment (eaddingtonwhite) [reply src/momento/aio/_scs_data_client.py:241]: Made issue to revisit. Agree @kvcache

Comment (kvcache) [reply src/momento/aio/_scs_data_client.py:241]: you can consider declining to propagate this bad pattern by merging [this fix](https://github.com/momentohq/client-sdk-python/pull/102) and updating your net new logging statements

Comment (kvcache) [reply src/momento/aio/_scs_data_client.py:241]: @malandis Did you miss this thread?

File: src/momento/simple_cache_client.py (hunk 2 above)
Comment (kvcache) [reply src/momento/simple_cache_client.py:290]: @malandis did you miss this as well?

**Pull request merged**